### ISSUE TITLE
Introduce standard gem ?

### ISSUE THREAD

**Issue by ericproulx** [ruby-grape/grape#2446]
There's a lot of traction around [standard](https://github.com/standardrb/standard?tab=readme-ov-file#usage). I think it would be a great idea to follow it but some work must be done before.

Any thoughts ?

**Comment by pieterocp** [comment]
Before creating a PR of this, thought I'd post a the preview pull request in here before creating a draft:

https://github.com/ruby-grape/grape/compare/master...pieterocp:grape:preview-standard?expand=1

Went with the latest version of standard, version 1.5.0, haven't tested the library with all the versions of ruby, but generally looks okayish?

(did have to make a couple of changes to run with rubocop upgrades to 1.75.5, so if that is something worth pulling out into it's own pr then happy to do that)

**Comment by dblock** [comment]
Rubocop changes are definitely worth pulling out in a separate PR. 

What's the story between `'` and `"`?

**Comment by pieterocp** [comment]
> Rubocop changes are definitely worth pulling out in a separate PR.

Pulled out with #2571 and merged, so thank you for that.

> What's the story between ' and "?

I think they've chosen in base settings to go with double quotes on string literals vs the currently enforced by `rubocop-1.75.8/config/default.yml` of `single_quotes`.

https://docs.rubocop.org/rubocop/cops_style.html#stylestringliterals

https://github.com/standardrb/standard/blob/ebfb1ddb2a60fd5870cfd68e9f7421e3f64c1763/config/base.yml#L1802C1-L1802C21
```yaml
Style/StringLiterals:
  Enabled: true
  EnforcedStyle: double_quotes
  ConsistentQuotesInMultiline: false
```

I am personally happy with the settings as they are now, having a look at the output from `git log -p .rubocop.yml` there doesn't seem to be too many changes but it's one for people who review and see friction to make a judgement on if they think it'd make life easier.

**Comment by pieterocp** [comment]
> So what's next for this issue?

Either, we collectively accept the standard gem into this project, fix every error and hand over control of RuboCop to the maintainers of standard gem, which does reduce any chance of bikeshedding.

Or we park this/close if we don't like the diff and keep control of configuration inside the project.

**Comment by ericproulx** [comment]
Hey @pieterocp, I'm working on the next major release (3.0) and it might be a good timing to do the ground work. About the `single quote` vs `double quote`, I don't mind having just the `"`.
